### PR TITLE
Add simple turn restrictions

### PR DESCRIPTION
**Pull request by dnesbitt61**
screenshot from 2015-02-18 14 23 55

Simple turn restrictions (one edge to another) are now handled. The left turn from McGovern Ave to Manheim Ave (Jordan's Curve) are properly avoided.

### CODE REVIEW

**Comment by dgearhart** [comment]
:+1:

File: src/thor/pathalgorithm.cc [hunk 1]
```diff
@@ -134,6 +134,7 @@ std::vector<GraphId> PathAlgorithm::GetBestPath(const PathLocation& origin,
   // Find shortest path
   uint32_t uturn_index, next_label_index;
   uint32_t prior_label_index;
+  uint32_t restriction;
   float dist2dest, dist;
   float cost, sortcost, currentcost;
   GraphId node;
```
Comment (kevinkreiser) [inline src/thor/pathalgorithm.cc]: we ought to some time un-c-ify some of these variables and move them into the loop. especially for the ones whose existance outside of the loops scope is meaningless, this is confusing, at least to me

**Comment by kevinkreiser** [comment]
:shipit:

**Comment by kevinkreiser** [comment]
google maps tab! for shame ;)

**Pull request merged**